test/prober.go

```diff
@@ -108,20 +108,37 @@ func (p *prober) handleResponse(response *spoof.Response) (bool, error) {
 		return p.stopped, nil
 	}
 
-	p.requests++
+	p.logRequestNoLock()
 	if response.StatusCode != http.StatusOK {
 		p.logf("%q status = %d, want: %d", p.url, response.StatusCode, http.StatusOK)
 		p.logf("response: %s", response)
 		p.failures++
 	}
-	if p.requests == p.minimumProbes {
-		close(p.minDoneCh)
-	}
 
 	// Returning (false, nil) causes SpoofingClient.Poll to retry.
 	return false, nil
 }
 
+func (p *prober) handleErrorRetry(err error) (bool, error) {
+	p.m.Lock()
+	defer p.m.Unlock()
+
+	p.logRequestNoLock()
+	p.failures++
+
+	// Returning true causes SpoofingClient.Poll to retry.
```

Comment:
This is the exact opposite of what is said above 🤔 

I assume this is because this is an error handler, and the other is a response handler?

Reply:
Yes. As in `type ErrorRetryChecker func(e error) (retry bool, err error)`, true is for retrying.